### PR TITLE
update(driver): add support to openat2 syscall

### PR DESCRIPTION
**Pull request by jasondellaluce**
```
Signed-off-by: Jason Dellaluce <jasondellaluce@gmail.com>
Co-authored-by: Federico Di Pierro <nierro92@gmail.com>
Co-authored-by: Grzegorz Nosek <grzegorz.nosek@sysdig.com>

```

**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area driver-kmod

/area driver-ebpf

/area libscap

/area libsinsp

**What this PR does / why we need it**:
This adds support for the `openat2` system call, along with its transformation flags.

**Which issue(s) this PR fixes**:
https://github.com/falcosecurity/falco/issues/676

**Does this PR introduce a user-facing change?**:

```release-note
update(driver): add support to openat2 syscall
```

### CODE REVIEW

**Comment by lucklypse** [comment]
Sorry for nitpicking but I think that it shouldn't say "Fixes falcosecurity/falco#676" otherwise poiana would close the issue if this PR is merged I think

File: userspace/libsinsp/examples/util.cpp [hunk 1]
```diff
@@ -132,7 +132,8 @@ std::string get_event_type(uint16_t type)
         case PPME_SYSCALL_OPENAT_E:
         case PPME_SYSCALL_OPENAT_2_E:
         case PPME_SYSCALL_OPENAT_X:
-        case PPME_SYSCALL_OPENAT_2_X: return "openat";
+        case PPME_SYSCALL_OPENAT_2_X:
+        case PPME_SYSCALL_OPENAT2_X: return "openat";
```
Comment (gnosek) [inline userspace/libsinsp/examples/util.cpp:136]: Shouldn't these be `openat2`?

**Review by gnosek — COMMENTED** [review]
LGTM! The event names are unfortunate since we have both OPENAT2_[EX] and OPENAT_2_[EX] but I can't see a way around this.

Also, I know openat2 has been added in 5.6 and there's no `struct open_how` before, but is there e.g. a `#define` we could use instead of checking the kernel version? Just wondering, but I expect RHEL to backport this syscall into some 2.6.32 or other :D Maybe even `#ifdef __NR_openat2` or similar? I rather like the idea of supporting the syscall even if the kernel does not, makes things simpler overall.

File: userspace/libsinsp/examples/util.cpp [hunk 2]
```diff
@@ -133,6 +133,7 @@ std::string get_event_type(uint16_t type)
         case PPME_SYSCALL_OPENAT_2_E:
         case PPME_SYSCALL_OPENAT_X:
         case PPME_SYSCALL_OPENAT_2_X: return "openat";
+        case PPME_SYSCALL_OPENAT2_X: return "openat2";
```
Comment (gnosek) [inline userspace/libsinsp/examples/util.cpp:136]: We probably want the enter event here too, right?

```suggestion
        case PPME_SYSCALL_OPENAT2_E:
        case PPME_SYSCALL_OPENAT2_X: return "openat2";
```

File: userspace/libsinsp/examples/util.cpp (hunk 1 above)
Comment (jasondellaluce) [reply userspace/libsinsp/examples/util.cpp]: Great catch! I fixed this in the latest commit.

File: driver/bpf/fillers.h [hunk 3]
```diff
@@ -2298,6 +2298,83 @@ FILLER(sys_openat_x, true)
...
+{
+	unsigned long resolve;
+	unsigned long flags;
+	unsigned long val;
+	unsigned long mode;
+	long retval;
+	int res;
+#ifdef __NR_openat2
+	struct open_how how;
+#endif
```
Comment (gnosek) [inline driver/bpf/fillers.h:2311]: Thanks for changing this. Did you get a chance to do a test build with an older (pre 5.6.0) kernel?

Comment (jasondellaluce) [reply driver/bpf/fillers.h:2311]: Thank you!

Comment (jasondellaluce) [reply driver/bpf/fillers.h:2311]: Thank you for your suggestions! I successfully built this on kernel version 5.4.0.

**Review by gnosek — APPROVED** [review]

**Review by leogr — APPROVED** [review]
/approve

**Review by mstemm — DISMISSED** [review]

**Comment by leogr** [comment]
Hey @jasondellaluce 

Could you rebase, please?

Thanks in advance :pray:

**Review by leogr — APPROVED** [review]
approved again

**Review by gnosek — APPROVED** [review]

**Pull request merged**